manifests/server/user.pp

```diff
@@ -191,36 +191,36 @@
 		default => '',
 	}
 
-	$args05 = type($initials) ? {
+	$args05 = type3x($initials) ? {
 		'string' => "--initials='${displayname}'",
 		'boolean' => $initials ? {
 			false => '',
 			# NOTE: [0,1] is a version robust way to get index 0...
-			default => sprintf("--initials='%s'", inline_template('<%= first[0,1]+last[0,1] %>')),
+                        default => "--initials='${first[0,1]}${last[0,1]}'",
 		},
 		default => '',
 	}
 
 	# email can provide a sensible default
 	$default_email_domain = $ipa::server::default_email_domain
-	$valid_email = type($email) ? {
+	$valid_email = type3x($email) ? {
 		'string' => "${email}" ? {
-			'' => [],	# assume managed but empty (rm values)
-			default => ["${email}"],
+			'' => '',	# assume managed but empty (rm values)
+			default => "${email}",
 		},
 		'array' => $email,
 		'boolean' => $email ? {
 			false => '',	# unmanaged
-			default => ["${valid_login}@${default_email_domain}"],	# sensible default
+			default => "${valid_login}@${default_email_domain}",	# sensible default
 		},
 		default => '',	# unmanaged
 	}
-	$args06 = type($valid_email) ? {
-		'array' => inline_template('<% if valid_email == [] %>--email=<% else %><%= valid_email.map {|x| "--email=\'"+x+"\'" }.join(" ") %><% end %>'),
+	$args06 = type3x($valid_email) ? {
```

Comment:
You're right, but that goes for a lot of parameters. According to the docs (ipa user-add --help) email should be string.
  --email=STR           Email address